include/bluetooth/services/ancs_client.h

```diff
@@ -0,0 +1,578 @@
+	BT_ANCS_EVENT_ID_NOTIFICATION_MODIFIED,
+	/** The iOS notification was removed. */
+	BT_ANCS_EVENT_ID_NOTIFICATION_REMOVED
+};
+
+/**@brief Control point command IDs that the Notification Consumer can send
+ *        to the Notification Provider.
+ */
+enum bt_ancs_cmd_id_val {
+	/** Requests attributes to be sent from the NP to the NC for a given
+	 *  notification.
+	 */
+	BT_ANCS_COMMAND_ID_GET_NOTIF_ATTRIBUTES,
+	/** Requests attributes to be sent from the NP to the NC for a given
+	 *  iOS app.
+	 */
+	BT_ANCS_COMMAND_ID_GET_APP_ATTRIBUTES,
+	/** Requests an action to be performed on a given notification.
+	 *  For example, dismiss an alarm.
+	 */
+	BT_ANCS_COMMAND_ID_GET_PERFORM_NOTIF_ACTION
+};
+
+/**@brief IDs for actions that can be performed for iOS notifications. */
+enum bt_ancs_action_id_values {
+	/** Positive action. */
+	ACTION_ID_POSITIVE = 0,
+	/** Negative action. */
+	ACTION_ID_NEGATIVE
+};
+
+/**@brief App attribute ID values.
+ * @details Currently, only one value is defined. However, the number of app
+ * attributes might increase. For this reason, they are stored in an enumeration.
+ */
+enum bt_ancs_app_attr_id_val {
+	/** Command used to get the display name for an app identifier. */
+	BT_ANCS_APP_ATTR_ID_DISPLAY_NAME = 0
+};
+
+/**@brief IDs for iOS notification attributes. */
+enum bt_ancs_notif_attr_id_val {
+	/** Identifies that the attribute data is of an "App Identifier" type. */
+	BT_ANCS_NOTIF_ATTR_ID_APP_IDENTIFIER = 0,
+	/** Identifies that the attribute data is a "Title". */
+	BT_ANCS_NOTIF_ATTR_ID_TITLE,
+	/** Identifies that the attribute data is a "Subtitle". */
+	BT_ANCS_NOTIF_ATTR_ID_SUBTITLE,
+	/** Identifies that the attribute data is a "Message". */
+	BT_ANCS_NOTIF_ATTR_ID_MESSAGE,
+	/** Identifies that the attribute data is a "Message Size". */
+	BT_ANCS_NOTIF_ATTR_ID_MESSAGE_SIZE,
+	/** Identifies that the attribute data is a "Date". */
+	BT_ANCS_NOTIF_ATTR_ID_DATE,
+	/** The notification has a "Positive action" that can be executed associated with it. */
+	BT_ANCS_NOTIF_ATTR_ID_POSITIVE_ACTION_LABEL,
+	/** The notification has a "Negative action" that can be executed associated with it. */
+	BT_ANCS_NOTIF_ATTR_ID_NEGATIVE_ACTION_LABEL
+};
+
+/**@brief Flags for iOS notifications. */
+struct bt_ancs_notif_flags {
+	/** If this flag is set, the notification has a low priority. */
+	uint8_t silent : 1;
+	/** If this flag is set, the notification has a high priority. */
+	uint8_t important : 1;
+	/** If this flag is set, the notification is pre-existing. */
+	uint8_t pre_existing : 1;
+	/** If this flag is set, the notification has a positive action that can be taken. */
+	uint8_t positive_action : 1;
+	/** If this flag is set, the notification has a negative action that can be taken. */
+	uint8_t negative_action : 1;
+};
+
+/**@brief Parsing states for received iOS notification and app attributes. */
+enum bt_ancs_parse_state {
+	/** Parsing the command ID. */
+	COMMAND_ID,
```

Comment:
These values should have BT_ANCS_(PARSE_STATE) as prefix, to avoid name collisions.

Reply:
Done.

The bt_ancs_action_id_values gets similar fix.

Reply:
The bt_ancs_parse_state and bt_ancs_action_id_values enum have BT_ANCS_ prefix added.

The bt_ancs_cmd_id_val BT_ANCS_COMMAND_ID_GET_PERFORM_NOTIF_ACTION is a typo. It should be BT_ANCS_COMMAND_ID_PERFORM_NOTIF_ACTION.